src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    transform(stateCopy, action);
```

Comment:
The `transform` function modifies `stateCopy`, but since `stateCopy` is not re-created for each action, the changes accumulate across actions. Move the creation of `stateCopy` inside the loop to fix this.